### PR TITLE
C#: Add missing `Transform{2D,3D}` and `Basis` constructors

### PR DESCRIPTION
**Pull request by raulsntos**
- Remove `Transform3D(Quaternion, Vector3)` constructor from C#.
	- Added to C# in https://github.com/godotengine/godot/pull/11899.
- Add `Transform3D(Projection)` constructor to C#.
	- Added to Core in https://github.com/godotengine/godot/pull/63219.
- Add documentation to the `Transform3D(Projection)` constructor in Core.
- Add `Transform3D` constructor with only real_t params to C# that mirrors `Transform2D`.
- Expose `Basis` constructor with only real_t params in C#.
- Add `Transform2D(real_t, Vector2, real_t, Vector2)` constructor to C#.
	- Added to Core in https://github.com/godotengine/godot/pull/52398.

### CODE REVIEW

**Comment by akien-mga** [comment]
> - Add Transform3D(Quaternion, Vector3) constructor to Core.
>   * Added to C# in #11899.

I think this was added by mistake / without approval from the core team, so I would remove the C# one that nobody ever requested instead.

See context: https://github.com/godotengine/godot/pull/11899/commits/676710f333db2853f103442beade3d416cfb1c7e#r143530753

File: doc/classes/Transform3D.xml
Comment (kleonc) [inline doc/classes/Transform3D.xml:44]: A little more detailed:
```suggestion
				Constructs a Transform3D from a [Projection] by trimming the last row of the projection matrix ([code]from.x.w[/code], [code]from.y.w[/code], [code]from.z.w[/code] and [code]from.z.w[/code] are not copied over).
```
+the same for the C# version.

File: doc/classes/Transform3D.xml
Comment (kleonc) [inline doc/classes/Transform3D.xml:44]: Sorry, a typo from my previous suggestion:
```suggestion
				Constructs a Transform3D from a [Projection] by trimming the last row of the projection matrix ([code]from.x.w[/code], [code]from.y.w[/code], [code]from.z.w[/code], and [code]from.w.w[/code] are not copied over).
```

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Transform3D.cs
Comment (kleonc) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Transform3D.cs:395]:
```suggestion
        /// <c>projection.y.w</c>, <c>projection.z.w</c>, and <c>projection.w.w</c>
```

**Review by kleonc — CHANGES_REQUESTED** [review]

File: doc/classes/Transform3D.xml
Comment (raulsntos) [reply doc/classes/Transform3D.xml]: Sorry, I totally missed it.

**Review by kleonc — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**
